backend/src/bundles/users/user.repository.ts

```diff
@@ -32,16 +33,32 @@ class UserRepository implements Repository {
 
     public async create(entity: UserEntity): Promise<UserEntity> {
         const { email, passwordHash } = entity.toNewObject();
-        const user = await this.userModel
-            .query()
-            .insert({
-                email,
-                passwordHash,
-            })
-            .returning('*')
-            .execute();
+        const trx = await this.userModel.startTransaction();
+
+        try {
+            const user = await this.userModel
+                .query(trx)
+                .insert({
+                    email,
+                    passwordHash,
+                })
+                .returning('*')
+                .execute();
+
+            const userDetails = await user
+                .$relatedQuery('userDetails', trx)
+                .insert({ userId: user.id });
 
-        return UserEntity.initialize({ ...user, fullName: null });
+            await trx.commit();
+
+            return UserEntity.initialize({
+                ...user,
+                fullName: userDetails.fullName,
+            });
+        } catch (error) {
+            await trx.rollback();
+            throw error;
```

Comment:
Should we use `throw new HttpError` here ?

Reply:
Oh, yes. Thank you

Reply:
Oh, wait. Actually, it can be handled by `initErrorHandler` inside `base-server-app.ts`. It returns this:
```
                const response: ServerCommonErrorResponse = {
                    errorType: ServerErrorType.COMMON,
                    message: error.message,
                };

                return replay
                    .status(HttpCode.INTERNAL_SERVER_ERROR)
                    .send(response);
```

Reply:
and this is exactly what we need, because if transaction fails this means there are some problems with the database -> server -> internal server error -> status code 500